Command/MigrateCommand.php

```diff
@@ -190,136 +146,37 @@ protected function execute(InputInterface $input, OutputInterface $output)
         
         $indexInDefaultEnv = true;
         if (strcmp($defaultEnv->getAlias(), $elasticsearchIndex) === 0 && strcmp($contentTypeNameFrom, $contentTypeNameTo) === 0) {
-            if (!$input->getOption('force')) {
+            if (!$forceImport) {
                 $output->writeln("<error>You can not import a content type on himself with the --force option</error>");
                 exit;
             }
             $indexInDefaultEnv = false;
         }
 
-        //https://www.elastic.co/guide/en/elasticsearch/client/php-api/current/_search_operations.html#_scrolling
         $arrayElasticsearchIndex = $this->client->search([
                 'index' => $elasticsearchIndex,
                 'type' => $contentTypeNameFrom,
                 'size' => $scrollSize,
                 "scroll" => $scrollTimeout,
                 'body' => $searchQuery,
         ]);
-        
-        $total = $arrayElasticsearchIndex["hits"]["total"];
-        // create a new progress bar
-        $progress = new ProgressBar($output, $total);
-        // start and displays the progress bar
+
+        $progress = new ProgressBar($output, $arrayElasticsearchIndex["hits"]["total"]);
         $progress->start();
-        
+        $importer = $this->importService->initDocumentImporter($contentTypeTo, 'SYSTEM_MIGRATE', $rawImport, $signData, $indexInDefaultEnv, $bulkSize, $finalize, $forceImport);
```

Comment:
Create a context object for all parameters that are passed in the initDocumentImporter. Then remove this function and pass the context to the import command itself. Your service should be stateless!

Reply:
The service is stateless. The function initDocumentImporter  is a factory

Reply:
1e8555d195f4caf512272f5f4efed1125cc64771